activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -56,19 +56,21 @@ def self.initialize_i18n(app)
       # Restore available locales check so it will take place from now on.
       I18n.enforce_available_locales = enforce_available_locales
 
-      directories = watched_dirs_with_extensions(reloadable_paths)
-      reloader = app.config.file_watcher.new(I18n.load_path.dup, directories) do
-        I18n.load_path.keep_if { |p| File.exist?(p) }
-        I18n.load_path |= reloadable_paths.flat_map(&:existent)
+      unless app.config.cache_classes
+        directories = watched_dirs_with_extensions(reloadable_paths)
+        reloader = app.config.file_watcher.new(I18n.load_path.dup, directories) do
+          I18n.load_path.keep_if { |p| File.exist?(p) }
```

Comment:
Don't we need to execute those when `cache_classes` is false? I think this is what `reloader.execute` was doing.

Reply:
You're right, to keep the same behavior as before we need to execute that code eagerly.

Won't be hard.